src/libcore/hash/mod.rs

```diff
@@ -603,6 +603,13 @@ mod impls {
         }
     }
 
+    #[stable(feature = "never_hash", since = "1.29.0")]
+    impl Hash for ! {
+        fn hash<H: Hasher>(&self, _: &mut H) {
+            unreachable!()
```

Comment:
It doesn't seem _that_ important, but using `*self` instead of `unreachable!()` is more consistent with how other impls for `!` are defined.

Reply:
@varkor I was under the impression that impls for `!` which return a value of another type generally do `*self`, but ones which return unit do `unreachable!()`. Although this seems pretty arbitrary and I may have just misremembered what I saw of existing impls.

I can change it to `*self` if that seems more reasonable.

Reply:
All the implementations I can see in libstd use `*self`, though there are a couple in rustc that use other methods. I don't think it's hugely important though — it's fairly obvious what's going on with either method.

Reply:
I like `*self` just so that it's trivial to translate -- no need to emit a bunch of panic call HIR+MIR+IR only for LLVM to figure out that it's unneeded again and remove it all.